src/lib/core/fiber.h

```diff
@@ -281,6 +281,27 @@ fiber_yield(void);
 API_EXPORT void
 fiber_start(struct fiber *callee, ...);
 
+/**
+ * Set an argument for the fiber function. Can be used to avoid calling
+ * fiber_start which means no yields.
+ *
+ * \param f     fiber to set the argument for
+ * \param f_arg argument for the fiber function
+ */
+API_EXPORT void
+fiber_set_arg(struct fiber *f, void *f_arg);
```

Comment:
I would think of a better name like `fiber_set_data` or `fiber_set_ctx`, because this function and its `get` version both are allowed to work even after the fiber is already started, apparently. We don't use that internally, but on the outside it more looks like "context" than "argument" at the moment. I don't mind the current name too, up to you.

Reply:
changed to fiber_set_ctx